lib/buffer.js

```diff
@@ -289,6 +289,24 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
+Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+  if (byteOffset > 0xffffffff)
+    byteOffset = 0xffffffff;
+  else if (byteOffset < 0)
+    byteOffset = 0;
+  byteOffset = byteOffset >>> 0;
+
+  if (typeof val === 'string')
+    return internal.indexOfString(this, val, byteOffset);
+  if (val instanceof Buffer)
+    return internal.indexOfBuffer(this, val, byteOffset);
```

Comment:
Minor improvement: `if (val.length === 0) return 0;` - skips an unnecessary transition to C++ land.